pkg/cloud/openstack/clients/machineservice.go

```diff
@@ -269,18 +273,64 @@ func (is *InstanceService) InstanceCreate(name string, config *openstackconfigv1
 		}
 	}
 	userData := base64.StdEncoding.EncodeToString([]byte(cmd))
-	createOpts = servers.CreateOpts{
+	clusterTag := ""
+	for _, tag := range config.Tags {
+		if strings.Contains(tag.Name, "ClusterID") {
+			clusterTag = tag.Name + tag.Value
+			break
+		}
+	}
+
+	var ports_list []servers.Network
+	for _, net := range nets {
+		// create server port
+		portCreateOpts := ports.CreateOpts{
+			Name:           name,
+			NetworkID:      net.UUID,
+			SecurityGroups: &config.SecurityGroups,
```

Comment:
Why is that? you can also pass here the SG name, right?

Reply:
This is how neutron works. https://developer.openstack.org/api-ref/network/v2/?expanded=create-port-detail

When passing SGs to servers.Create, OpenStack takes care of the name <=> uuid translation. When creating ports directly, only uuids are supported.

Btw. When using `openstack --debug port create --network net-test --security-group test-group blahfasel`, the OpenStack client fetches all SecurityGroups and does the translation, and then posts to the ports endpoint with only uuids included.

I would be totally fine to implement this the same way.

Reply:
But that should be handled on the gopghercloud side, isn't it? And based on the testing examples it seems that it is already handled:
https://github.com/gophercloud/gophercloud/blob/master/openstack/networking/v2/ports/testing/requests_test.go#L185-L204

Reply:
Hmm. Seems this is not the case. I am currently only able to test against OpenStack Pike... 

I tested with the code below against an OpenStack Pike, and get `'kube-group' is not a valid UUID`.

I think, because the official openstack cli also converts names to UUID before creating the port, I would argue that this is not anything `gophercloud` should do, but the enduser should. It would be also fine if the enduser is the one who defines the yaml. But then it must be documented prominently.

For me the test mentioned test should be changed.

```
package main

import (
        "flag"
        "fmt"
        "log"
        "os"

        "github.com/gophercloud/gophercloud"
        "github.com/gophercloud/gophercloud/openstack"
        "github.com/gophercloud/gophercloud/openstack/networking/v2/ports"
        "github.com/gophercloud/utils/openstack/clientconfig"
)

func getNetworkClient() (*gophercloud.ServiceClient, error) {
        opts, err := clientconfig.AuthOptions(&clientconfig.ClientOpts{})
        if err != nil {
                return nil, err
        }

        provider, err := openstack.AuthenticatedClient(*opts)
        if err != nil {
                return nil, fmt.Errorf("create providerClient err: %v", err)
        }

        client, err := openstack.NewNetworkV2(provider, gophercloud.EndpointOpts{})
        if err != nil {
                return nil, err
        }

        return client, nil
}


func main() {
        networkID := flag.String("net-id", "", "a valid network id")
        groupName := flag.String("security-group-name", "", "a valid security group name")
        groupID := flag.String("security-group-id", "", "a valid security group id")

        flag.Parse()

        if *networkID == "" {
                log.Fatal("Please provide a net-id")
        }
        if *groupName == "" {
                log.Fatal("Please provide a security-group-name")
        }
        if *groupID == "" {
                log.Fatal("Please provide a security-group-id")
        }

        client, err := getNetworkClient()
        if err != nil {
                log.Fatalf("Unable to get network client: %v", err)
        }

        port, err := ports.Create(client, ports.CreateOpts{
                Name:      "test-port-with-uuid",
                NetworkID: *networkID,
                SecurityGroups: &[]string{
                        *groupID,
                },
        }).Extract()
        if err != nil {
                log.Fatalf("Unable to create port with UUID in SecurityGroups: %v", err)
        }
        fmt.Printf("Created port with UUID in SecurityGroups: %v\n", port)

        port, err = ports.Create(client, ports.CreateOpts{
                Name:      "test-port-with-name",
                NetworkID: *networkID,
                SecurityGroups: &[]string{
                        *groupName,
                },
        }).Extract()

        if err != nil {
                log.Fatalf("Unable to create port with Name in SecurityGroups: %v", err)
        }

        fmt.Printf("Created port with Name in SecurityGroups: %v\n", port)
}
```

```
$ dep status
PROJECT                             CONSTRAINT     VERSION        REVISION  LATEST   PKGS USED
github.com/gophercloud/gophercloud  branch master  branch master  a9f9006   a9f9006  8
github.com/gophercloud/utils        branch master  branch master  34f5991   34f5991  1
gopkg.in/yaml.v2                    v2.2.2         v2.2.2         51d6538   v2.2.2   1

$ go run main.go -net-id=92cbf083-e057-4974-b24d-dce55279f4f9 -security-group-name=kube-group -security-group-id=f2c2ff1a-0785-4c23-b316-7d8c1b823b20

Created port with UUID in SecurityGroups: &{51e35d87-802d-4818-8c6f-0b300d153cd5 92cbf083-e057-4974-b24d-dce55279f4f9 test-port-with-uuid  true DOWN fa:16:3e:64:a9:c1 [{f44ee19c-bbe5-414a-81f1-34e323c5e37d 192.168.0.12}] bf8e5f9815c54afea53962fd4a13c745 bf8e5f9815c54afea53962fd4a13c745  [f2c2ff1a-0785-4c23-b316-7d8c1b823b20]  [] []}
2019/01/15 21:42:17 Unable to create port with Name in SecurityGroups: Bad request with: [POST https://network.fra.optimist.innovo.cloud/v2.0/ports], error message: {"NeutronError": {"message": "Invalid input for security_groups. Reason: 'kube-group' is not a valid UUID.", "type": "HTTPBadRequest", "detail": ""}}
exit status 1

$ openstack security group show kube-group
+-----------------+--------------------------------------+
| Field           | Value 
+-----------------+--------------------------------------+
| created_at      | 2018-10-25T14:38:07Z
| id              | f2c2ff1a-0785-4c23-b316-7d8c1b823b20
| name            | kube-group
```